File: backend/collectors/python-agent/log_collector.py
import requests
import time
import yaml
import logging

with open('config.yaml') as f:
    config = yaml.safe_load(f)

BACKEND_URL = config['backend_url']
INTERVAL = config.get('interval', 5)

logging.basicConfig(level=logging.INFO)

def collect_logs():
    return {
        "level": "INFO",
        "message": "Python agent log",
        "source": "python-agent",
        "timestamp": time.strftime("%Y-%m-%dT%H:%M:%S")
    }

def send_log(log):
    try:
        r = requests.post(BACKEND_URL, json=log)
        if r.status_code in (200, 201):
            logging.info("Sent log")
        else:
            logging.error(r.text)
    except Exception as e:
        logging.error(e)

if __name__ == "__main__":
    while True:
        log = collect_logs()
        send_log(log)
        time.sleep(INTERVAL)

File: backend/collectors/java-agent/LogCollector.java
package com.clmas.collector;

import java.net.HttpURLConnection;
import java.net.URL;
import java.io.OutputStream;
import java.time.LocalDateTime;

public class LogCollector {

    public static void main(String[] args) throws Exception {
        while (true) {
            String json = String.format("{\"level\":\"INFO\",\"message\":\"Java agent log\",\"source\":\"java-agent\",\"timestamp\":\"%s\"}", LocalDateTime.now());
            sendLog(json);
            Thread.sleep(10000);
        }
    }

    public static void sendLog(String json) throws Exception {
        URL url = new URL("http://localhost:8080/logs");
        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
        conn.setRequestMethod("POST");
        conn.setRequestProperty("Content-Type", "application/json");
        conn.setDoOutput(true);

        try (OutputStream os = conn.getOutputStream()) {
            os.write(json.getBytes());
            os.flush();
        }
        System.out.println("Sent log: " + conn.getResponseCode());
        conn.disconnect();
    }
}
